aviatrix/resource_aviatrix_spoke_transit_attachment.go

```diff
@@ -50,6 +76,15 @@ func resourceAviatrixSpokeTransitAttachmentCreate(d *schema.ResourceData, meta i
 
 	attachment := marshalSpokeTransitAttachmentInput(d)
 
+	spoke, err := client.GetGateway(&goaviatrix.Gateway{GwName: attachment.SpokeGwName})
```

Comment:
Is this API call needed? Should we be validating this on Terraform side or should we be leaving it to the API/backend to validate? 

I have something similar in Private Mode where an attribute can only be used on Transit/Spoke Gateways if the Controller is in Private Mode, but it seems expensive to make every Spoke/Transit Gateway have that API call in create. Especially for non-Private Mode setups. In theory, create is only called once per lifetime for a Gateway, so the cost might not be that big.

What do you think @zhiqiangjin?

Reply:
Talked with Tim yesterday afternoon, let's keep it for now.